test/integration/puppetlabs/services/master/environment_modules_int_test.clj

```diff
@@ -0,0 +1,392 @@
+           ;;    the 'test' environment was not flushed.
+           (purge-env-dir)
+           (purge-all-env-caches)
+           ;; add version that has multiple module versions
+           (let [prod-file (testutils/write-pp-file "class oneprod {}"
+                                                    "somemod"
+                                                    "prod"
+                                                    "production"
+                                                    "1.0.0")
+                 test-file (testutils/write-pp-file "class onetest {}"
+                                                    "somemod"
+                                                    "test"
+                                                    "test"
+                                                    "2.0.0")
+                 production-response-initial (get-env-modules "production")
+                 test-response-initial (get-env-modules "test")]
+             (is (= 200 (:status production-response-initial))
+                 (str
+                  "unexpected status code for initial production response"
+                  "response: "
+                  (ks/pprint-to-string production-response-initial)))
+             (is (= {"modules" [{"name" "somemod", "version" "1.0.0"}]
+                     "name" "production"}
+                    (response->module-info-map production-response-initial))
+                 "unexpected body for production response")
+             (is (= 200 (:status test-response-initial))
+                 (str
+                  "unexpected status code for initial test response"
+                  "response: "
+                  (ks/pprint-to-string test-response-initial)))
+             (is (= {"modules"[{"name" "somemod", "version" "2.0.0"}]
+                     "name" "test"}
+                    (response->module-info-map test-response-initial))
+                 "unexpected body for test response")
+
+             (testutils/write-pp-file "class oneflushprod {}"
+                                      "somemod"
+                                      "prod"
+                                      "production"
+                                      "1.0.0")
+             (testutils/write-pp-file "class oneflushtest {}"
+                                      "somemod"
+                                      "test"
+                                      "test"
+                                      "2.0.0")
+
+             (let [production-response-before-flush (get-env-modules
+                                                     "production")
+                   test-response-before-flush (get-env-modules
+                                               "test")]
+               (is (= 304 (:status production-response-before-flush))
+                   (str
+                    "unexpected status code for prod response after code change "
+                    "but before flush"))
```

Comment:
Nah, also the analytics endpoint will be going through my JRuby wrapper function, and not making an HTTP request. And the request isn't so frequent that it needs to be cached. I'll remove these, then. Thanks!